Sources/NIO/EventLoopFuture.swift

```diff
@@ -844,9 +846,9 @@ extension EventLoopFuture {
     ///     - futures: An array of `EventLoopFuture<U>` to wait for.
     ///     - with: A function that will be used to fold the values of two `EventLoopFuture`s and return a new value wrapped in an `EventLoopFuture`.
     /// - returns: A new `EventLoopFuture` with the folded value whose callbacks run on `self.eventLoop`.
-    public func fold<U>(_ futures: [EventLoopFuture<U>], with combiningFunction: @escaping (T, U) -> EventLoopFuture<T>) -> EventLoopFuture<T> {
-        let body = futures.reduce(self) { (f1: EventLoopFuture<T>, f2: EventLoopFuture<U>) -> EventLoopFuture<T> in
-            let newFuture = f1.and(f2).then { (args: (T, U)) -> EventLoopFuture<T> in
+    public func fold<U>(_ futures: [EventLoopFuture<U>], with combiningFunction: @escaping (Value, U) -> EventLoopFuture<Value>) -> EventLoopFuture<Value> {
```

Comment:
Same question here about `U`.

Reply:
thanks, renamed to `NewValue`, `OtherValue`, or `InputValue` depending on the context.